package/src/components/Channel/Channel.tsx

```diff
@@ -927,8 +971,20 @@ const ChannelWithContext = <
 
       channel.state.clearMessages();
       channel.state.addMessagesSorted(finalMessages);
+
       setHasMore(true);
       copyChannelState();
+      await reloadThread();
+
+      if (failedMessages.length) {
```

Comment:
Only thought is  maybe we should do this above reloadThread so it's not hungup by the await or a failed reload, the failed threadMessages should be added after the reloadThread as they are now.